### PR TITLE
Feature Finder Sync Plugin for OS X 10.10

### PR DESCRIPTION
**Pull request by Chilledheart**
Support:
- Show Sync Icon
- Get Share Link in the menu

screen shot 2015-02-02 at 15 27 02

screen shot 2015-02-02 at 15 28 11

### CODE REVIEW

File: src/fsplugin-handler.mm [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+namespace {
+std::mutex watch_set_mutex_;
+std::vector<LocalRepo> watch_set_;
+QHash<QString, Account> accounts_cache_;
+const char *kRepoRelayAddrProperty = "relay-address";
+}
+
+Account findAccountByRepo(const QString& repo_id)
+{
```
Comment (lins05) [inline src/fsplugin-handler.mm]: we can move this function to SeafileApplet class, or utils.cpp.

Comment (Chilledheart) [inline src/fsplugin-handler.mm]: I am good with that.

**Comment by lins05** [comment]
Maybe we should use "Get Seafile Shared Link" in the menu? Otherwise the user may not know what "Get Share Link" does.

**Comment by Chilledheart** [comment]
Updated in https://github.com/haiwen/seafile-client/commit/cd1fa0655e47b1e6f929b58d6cd74dcc30ffa7a9

**Comment by Chilledheart** [comment]
TODO: add an option to enable/disable finder sync plugin

**Comment by Chilledheart** [comment]
DONE: add an option to enable/disable finder sync plugin

**Comment by Chilledheart** [comment]
Preview:
screen shot 2015-03-27 at 17 52 56

**Comment by Chilledheart** [comment]
Preview:

screen shot 2015-03-31 at 6 22 47 pm
screen shot 2015-03-31 at 6 21 59 pm

**Pull request merged**